Purchases/SubscriberAttributes/RCSubscriberAttributesManager.m

```diff
@@ -65,30 +66,60 @@ - (void)setPushTokenString:(nullable NSString *)pushTokenString appUserID:(NSStr
 
 - (void)syncIfNeededWithAppUserID:(NSString *)appUserID completion:(void (^)(NSError *_Nullable error))completion {
     if ([self numberOfUnsyncedAttributesForAppUserID:appUserID] > 0) {
-        [self syncAttributesWithCompletion:completion appUserID:appUserID];
+        [self syncAttributesWithAppUserID:appUserID completion:completion];
     } else {
         completion(nil);
     }
 }
 
+- (void)syncAttributesForAllUsersWithCurrentAppUserID:(NSString *)currentAppUserID {
+    NSDictionary < NSString * , RCSubscriberAttributeDict > *unsyncedAttributesForAllUsers =
+        [self unsyncedAttributesByKeyForAllUsers];
+
+    for (NSString *syncingAppUserID in unsyncedAttributesForAllUsers.allKeys) {
+        [self syncAttributes:unsyncedAttributesForAllUsers[syncingAppUserID]
+                forAppUserID:syncingAppUserID
+                  completion:^(NSError *error) {
+                      if (error == nil && syncingAppUserID != currentAppUserID) {
+                          [self.deviceCache deleteAttributesIfSyncedForAppUserID:syncingAppUserID];
+                          RCLog(@"Subscriber attributes synced successfully for appUserID: %@", syncingAppUserID);
+                      } else {
+                          RCErrorLog(@"error when syncing subscriber attributes. Details: %@\n UserInfo:%@",
```

Comment:
If error is nil but `syncingAppUserID == currentAppUserID` it will also print this error right?

Reply:
good catch. there was also a problem if where we wouldn't log success if the current app user id was the one being synced. fixed and extracted to a separate method